synapse/app/generic_worker.py

```diff
@@ -126,6 +126,11 @@
 
 logger = logging.getLogger("synapse.app.generic_worker")
 
+DIRECTORY_UPDATE_WARNING = """
+The user directory worker is not allowed to perform user directory updates per the config.
```

Comment:
right, I see the confusion here now.

"not allowed to" sounds like I've tried to enable something invalid.  Maybe `s/not allowed to/not configured to/`.

But I don't really understand why this is a thing we need to warn about. Why is it a problem if the updates are done on the main process?

Note that the current wording sounds very much like an error, indicating something that definitely won't work.

Reply:
> Maybe s/not allowed to/not configured to/.

👍

> Why is it a problem if the updates are done on the main process?

That's not the problem, the problem is is that the configuration does not appoint the local worker to do the updates.

The config can now appoint any worker to do this job by its name, so the job of the user directory worker isn't that unique anymore, but it's uniqueness is still supported by synapse.

> Note that the current wording sounds very much like an error

That's because it is, the local worker has a "type" of being a designated user update worker, while the configuration does not allow it to do just that. This is more or less a warning that "your config, as-is, is contradictory"

The previous version exited the process with `1` when hitting this point, should that be re-introduced?